### PR TITLE
misc: Add macros for Velox::Expected

### PR DESCRIPTION
**Pull request by marin-ma**
Add macros `VELOX_RETURN_UNEXPECTED_IF`, `VELOX_RETURN_UNEXPECTED_NOT_OK` and `VELOX_RETURN_UNEXPECTED`.

### CODE REVIEW

**Comment by marin-ma** [comment]
@pedroerp Could you help to review? Thanks!

File: velox/common/base/Status.h [hunk 1]
```diff
@@ -485,6 +485,29 @@ void Status::moveFrom(Status& s) {
     VELOX_RETURN_IF(!__s.ok(), __s);                          \
   } while (false)
 
+/// Return with given status wrapped in folly::Unexpected if condition is met.
```
Comment (pedroerp) [inline velox/common/base/Status.h:488]: can we also add some examples in the documentation on the top of the file?

**Review by pedroerp — APPROVED** [review]
Thanks you. Please just add a few examples in the header docs.

File: velox/common/base/Status.h (hunk 1 above)
Comment (marin-ma) [reply velox/common/base/Status.h]: @pedroerp Added the doc. Please help to review again. Thanks!

File: velox/common/base/Status.h [hunk 2]
```diff
@@ -83,6 +83,49 @@ namespace facebook::velox {
 ///  VELOX_RETURN_NOT_OK(operation2());
 ///  VELOX_RETURN_NOT_OK(operation3());
 ///  ...
+///
+/// folly::Expected<T, Status> holds either a result or an error status and is
+/// aliased as Expected<T> to simplify usage.
+///
+/// For operations return a value if successful, or error status if not,
```
Comment (pedroerp) [inline velox/common/base/Status.h:90]: For operations that return ...

**Review by pedroerp — APPROVED** [review]
Thank you @marin-ma

**Pull request closed without merging**